test/logoutresponse_test.rb

```diff
@@ -30,36 +26,125 @@ class RubySamlTest < Minitest::Test
 
-        assert logoutresponse.validate
+        assert logoutresponse.validate!(true)
+
+        assert_equal in_relation_to_request_id, logoutresponse.in_response_to
+
+        assert logoutresponse.success?
+        assert_empty logoutresponse.errors
+      end
+
+      it "validate the response extended" do
+        in_relation_to_request_id = random_id
+        custom_settings = settings
+        custom_settings.idp_entity_id = 'http://app.muda.no'
+
+        logoutresponse = OneLogin::RubySaml::Logoutresponse.new(valid_response({:uuid => in_relation_to_request_id}), custom_settings)
+
+        assert logoutresponse.validate!(true, in_relation_to_request_id)
 
-        assert_equal settings.issuer, logoutresponse.issuer
         assert_equal in_relation_to_request_id, logoutresponse.in_response_to
 
         assert logoutresponse.success?
+        assert_empty logoutresponse.errors
+      end
+
+      it "invalidate response when initiated with blank" do
+
+        logoutresponse = OneLogin::RubySaml::Logoutresponse.new("", settings)
+
+        assert !logoutresponse.validate!(true)
+        assert logoutresponse.errors.include? "Blank Logout Response"
+      end
+
+      it "invalidate response when initiated with no idp cert or fingerprint" do
+        bad_settings = settings
+        bad_settings.idp_cert_fingerprint = nil
+        bad_settings.idp_cert = nil
+        logoutresponse = OneLogin::RubySaml::Logoutresponse.new(valid_response, bad_settings)
+        assert !logoutresponse.validate!(true)
+        assert logoutresponse.errors.include? "No fingerprint or certificate on settings"
       end
 
       it "invalidate responses with wrong id when given option :matches_uuid" do
 
         expected_request_id = "_some_other_expected_uuid"
-        opts = { :matches_request_id => expected_request_id}
 
-        logoutresponse = OneLogin::RubySaml::Logoutresponse.new(valid_response, settings, opts)
+        logoutresponse = OneLogin::RubySaml::Logoutresponse.new(valid_response, settings)
 
-        assert !logoutresponse.validate
+        assert !logoutresponse.validate!(true, expected_request_id)
         refute_equal expected_request_id, logoutresponse.in_response_to
       end
 
       it "invalidate responses with wrong request status" do
         logoutresponse = OneLogin::RubySaml::Logoutresponse.new(unsuccessful_response, settings)
 
-        assert !logoutresponse.validate
+        assert !logoutresponse.validate!(true)
+        assert !logoutresponse.success?
+        assert logoutresponse.errors.include? "The status code of the Logout Response was not Success, was Requester"
+      end
+
+      it "invalidate responses with wrong issuer" do
+        in_relation_to_request_id = random_id
+        bad_settings = settings
+        bad_settings.idp_entity_id = 'http://invalid.issuer.example.com/'
+        logoutresponse = OneLogin::RubySaml::Logoutresponse.new(valid_response({:uuid => in_relation_to_request_id}), bad_settings)
```

Comment:
Same comment about `valid_response` as above. Also applies to the lines below too.